spec/services/shared_examples_preserved_object_handler.rb

```diff
@@ -119,53 +119,81 @@
-    po_handler.send(method_sym)
-    expect(pc.reload.size).to eq 1
-  end
-  it 'does not update PreservedCopy last_audited field' do
-    orig_timestamp = pc.last_audited
-    po_handler.send(method_sym)
-    expect(pc.reload.last_audited).to eq orig_timestamp
-  end
-  it 'does not update PreservedCopy last_checked_on_storage' do
-    orig_timestamp = pc.last_checked_on_storage
-    po_handler.send(method_sym)
-    expect(pc.reload.last_checked_on_storage).to eq orig_timestamp
+  context 'PreservedCopy' do
+    context 'changed' do
+      it 'last_version_audit' do
+        orig = pc.last_version_audit
+        po_handler.send(method_sym)
+        expect(pc.reload.last_version_audit).to be > orig
+      end
+      if method_sym == :update_version
+        it 'status becomes EXPECTED_VERS_NOT_FOUND_ON_STORAGE_STATUS' do
+          orig = pc.status
+          po_handler.send(method_sym)
+          expect(pc.reload.status).to eq PreservedCopy::EXPECTED_VERS_NOT_FOUND_ON_STORAGE_STATUS
+          expect(pc.status).not_to eq orig
+        end
+      end
+    end
+    context 'unchanged' do
+      it "version" do
+        orig = pc.version
+        po_handler.send(method_sym)
+        expect(pc.reload.version).to eq orig
+      end
+      it "size" do
+        orig = pc.size
+        po_handler.send(method_sym)
+        expect(pc.reload.size).to eq orig
+      end
+      it 'last_moab_validation' do
+        orig = pc.last_moab_validation
+        po_handler.send(method_sym)
+        expect(pc.reload.last_moab_validation).to eq orig
+      end
+      if method_sym != :update_version
+        it 'status becomes EXPECTED_VERS_NOT_FOUND_ON_STORAGE_STATUS' do
+          orig = pc.status
+          po_handler.send(method_sym)
+          expect(pc.status).to eq orig
+        end
+      end
+    end
   end
-  it 'does not update status of PreservedCopy' do
-    orig_status = pc.status
-    po_handler.send(method_sym)
-    expect(pc.reload.status).to eq orig_status
+  context 'PreservedObject' do
+    context 'unchanged' do
+      it "PreservedObject current_version stays the same" do
+        pocv = po.current_version
+        po_handler.send(method_sym)
+        expect(po.reload.current_version).to eq pocv
+      end
+    end
   end
+
   it "logs at error level" do
     expect(Rails.logger).to receive(:log).with(Logger::ERROR, unexpected_version_msg)
     expect(Rails.logger).not_to receive(:log).with(Logger::INFO, updated_pc_db_timestamp_msg)
     expect(Rails.logger).not_to receive(:log).with(Logger::ERROR, updated_po_db_timestamp_msg)
-    expect(Rails.logger).not_to receive(:log).with(Logger::INFO, updated_status_msg_regex)
+    allow(Rails.logger).to receive(:log).with(Logger::INFO, updated_status_msg_regex)
     po_handler.send(method_sym)
   end
 
   context 'returns' do
     let!(:results) { po_handler.send(method_sym) }
+    let(:num_results) do
+      if method_sym == :update_version
+        5
+      else
+        4
+      end
+    end
```

Comment:
does this actually get used in this context?  i can see where the context that starts on line 305 uses the `num_results` it defines, but this one looks unused to me.

Reply:
again, good :eyes:.  Removed.